File: hotel-server/src/main/java/eu/getsoftware/hotelico/service/booking/application/customer/domain/infrastructure/security/UserContext.java
// path: hotelico/service/booking/application/customer/domain/infrastructure/security/UserContext.java

package eu.getsoftware.hotelico.service.booking.application.customer.domain.infrastructure.security;

import org.springframework.security.core.GrantedAuthority;

import java.util.List;

public class UserContext {

	private final String username;
	private final List<GrantedAuthority> authorities;

	public UserContext(String username, List<GrantedAuthority> authorities) {
		this.username = username;
		this.authorities = authorities;
	}

	public String getUsername() {
		return username;
	}

	public List<GrantedAuthority> getAuthorities() {
		return authorities;
	}

	public static UserContext create(String username, List<GrantedAuthority> authorities) {
		if (username == null || username.isBlank()) {
			throw new IllegalArgumentException("Username is blank: " + username);
		}
		return new UserContext(username, authorities);
	}
}

File: hotel-app/src/main/kotlin/eu/getsoftware/hotelico/service/booking/application/customer/domain/infrastructure/security/UserContext.kt
package eu.getsoftware.hotelico.service.booking.application.customer.domain.infrastructure.security

import org.springframework.security.core.GrantedAuthority

/**
 *
 *
 * Aug 4, 2016
 */
class UserContext {
	
	var username: String
	var authorities: List<GrantedAuthority>
	
	constructor(username: String, authorities: List<GrantedAuthority>) {
		this.username = username
		this.authorities = authorities
	}
	
	companion object {
		
		@JvmStatic
		fun create(username: String, authorities: kotlin.collections.List<GrantedAuthority>): eu.getsoftware.hotelico.service.booking.application.customer.domain.infrastructure.security.UserContext {
			if (username.isNullOrBlank()) throw IllegalArgumentException("Username is blank: $username")
			return eu.getsoftware.hotelico.service.booking.application.customer.domain.infrastructure.security.UserContext(
                username,
                authorities
            )
		}
	}
	
}
